datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -181,27 +185,37 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
 
                 auth = requests_ntlm.HttpNtlmAuth(config['ntlm_domain'], config['password'])
 
-        if auth is None and config['kerberos_auth']:
-            ensure_kerberos()
+        if auth is None:
+            if config['kerberos_auth']:
+                ensure_kerberos()
 
-            # For convenience
-            if is_affirmative(config['kerberos_auth']):
-                config['kerberos_auth'] = 'required'
+                # For convenience
+                if is_affirmative(config['kerberos_auth']):
+                    config['kerberos_auth'] = 'required'
 
-            if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
-                raise ConfigurationError(
-                    'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
-                        config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
+                if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
+                    raise ConfigurationError(
+                        'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
+                            config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
+                        )
                     )
+
+                auth = requests_kerberos.HTTPKerberosAuth(
+                    mutual_authentication=KERBEROS_STRATEGIES[config['kerberos_auth']],
+                    delegate=is_affirmative(config['kerberos_delegate']),
+                    force_preemptive=is_affirmative(config['kerberos_force_initiate']),
+                    hostname_override=config['kerberos_hostname'],
+                    principal=config['kerberos_principal'],
                 )
+            elif config['aws_host']:
```

Comment:
AFAICT `aws_host` is only required by aws-requests-host to create a `Host` header. We may be able to lift this dependency in the future by fetching the value of the `Host` header (or defaulting to urlparse.urlparse(url).hostname) by contributing upstream or by subclassing BotoAWSRequestsAuth.

Reply:
For now I think it's fine but maybe we can make the `aws_host` parameter optional in the future?

Reply:
That's the part I'm confused about, are we sure the host corresponds to the endpoint we're hitting?

Reply:
Seems like it:
https://docs.aws.amazon.com/general/latest/gr/sigv4_elements.html
https://docs.aws.amazon.com/general/latest/gr/rande.html

Also see this example where aws relies on `requests` creating this `Host` header automatically from the endpoint: https://docs.aws.amazon.com/general/latest/gr/sigv4-signed-request-examples.html